cfbs/commands.py

```diff
@@ -1000,3 +1001,74 @@ def input_command(args, input_from="cfbs input"):
         files_to_commit.append(input_path)
     config.save()
     return Result(0, do_commit, None, files_to_commit)
+
+
+def set_input_command(name, src_path):
+    config = CFBSConfig.get_instance()
+    module = config.get_module_from_build(name)
+    if module is None:
+        log.error("Module '%s' not found" % name)
+        return 1
+
+    spec = module.get("input")
+    if spec is None:
+        log.error("Module '%s' does not accept input" % name)
+        return 1
+    log.debug("Input spec for module '%s': %s" % (name, pretty(spec)))
+
+    if src_path is None:
+        try:
+            data = json.load(stdin)
+        except json.decoder.JSONDecodeError as e:
+            log.error("Error reading json from stdin: %s" % e)
+            return 1
+    else:
+        data = read_json(src_path)
+        if data is None:
+            log.error("Failed to read json from file '%s'" % src_path)
+            return 1
+
+    assert data is not None
+    log.debug("Input data for module '%s': %s" % (name, pretty(data)))
+
+    def _compare_dict(a, b, ignore={}):
+        assert isinstance(a, dict) and isinstance(b, dict)
+        if set(a.keys()) != set(b.keys()) - ignore:
+            return False
+        """Avoid code duplication by converting the values of the two dicts
+        into two lists in the same order and compare the lists instead"""
```

Comment:
Please use comments for what is commenting the code. Documentation strings are for documentation.